density_scaler/src/hex.rs

```diff
@@ -45,20 +45,29 @@ static HIP17_RES_CONFIG: [Option<HexResConfig>; 11] = [
     Some(HexResConfig::new(2, 1, 1)),     // 10
 ];
 
-#[derive(Debug, Serialize, Eq, PartialEq)]
-pub struct ScalingMap(HashMap<String, Decimal>);
+#[async_trait::async_trait]
+pub trait HexDensityMap: Clone {
+    async fn get(&self, hex: &str) -> Option<Decimal>;
+    async fn swap(&self, new_map: HashMap<String, Decimal>);
+}
+
+#[derive(Debug, Clone)]
+pub struct SharedHexDensityMap(Arc<RwLock<HashMap<String, Decimal>>>);
 
-impl ScalingMap {
+impl SharedHexDensityMap {
```

Comment:
perhaps outside of the scope of this PR, but I so new reason for this trait to exists. And having an `fn new()`  in a trait is almost never needed, if at all

Reply:
As for the `HexDensityMap` trait, I was thinking having the trait would help with testing later, so we could swap in a different implementation if need be.

As for the `new` function, I like how the code outside this module does not know about the Arc/RowLock

Is there a reason not to have the `new` function?

Reply:
> Is there a reason not to have the new function?

Sorry, my fault. I misread and thought the `HexDensityMap` trait had a `new`.